models/Recording.js

```diff
@@ -115,9 +115,43 @@ module.exports = function(sequelize, DataTypes) {
       offset: offset,
       attributes: this.userGetAttributes,
     };
-    return this.findAndCount(q);
+
+    var queryResponse = await this.findAndCount(q);
+    filterRecordings(user, queryResponse.rows, filterOptions);
+    return queryResponse;
   };
 
+  function filterRecordings(user, recordings, options = {}) {
+    if (typeof options.latLongAcc != 'number') {
+      options.latLongAcc = 100;
+    }
+    if (!user.hasGlobalWrite()) {
+      options.latLongAcc = Math.max(options.latLongAcc, 100);
+    }
+
+    for (var i in recordings) {
+      if (recordings[i].location) {
+        filterLatLong(recordings[i].location.coordinates, options.latLongAcc);
+      }
+    }
+  }
+
+  function filterLatLong(latLong, acc) {
+    const resolution = acc*360/40000000;
+    latLong[0] = latLong[0] - latLong[0]%resolution;
+    if (latLong[0] > 0) {
+      latLong[0] += resolution/2;
+    } else {
+      latLong[0] -= resolution/2;
+    }
+    latLong[1] = latLong[1] - latLong[1]%resolution;
+    if (latLong[1] > 0) {
+      latLong[1] += resolution/2;
+    } else {
+      latLong[1] -= resolution/2;
+    }
+  }
```

Comment:
How about:

```js

function reduceLatLonPrecision(latLon, prec) {
  assert (latLon.length == 2);

  const resolution = prec * 360 / 40000000;
  const half_resolution = resolution / 2;
  return latLon.map((val) => {
     val = val - (val % resolution);
     if (val > 0) {
        val += half_resolution;
     } else {
        val -= half_resolution;
     }
     return val;
  });
}
```
This avoid duplicating the code for lat and lon (less chance of bugs) and returning a new array instead of modifying the existing one in place reduces the chances for strange bugs (side effects are evil).

The call above would then change to:
```js
recordings[i].location.coordinates = reduceLatLonPrecision(recordings[i].location.coordinates, options.latLongAcc);
```